python/pyarrow/parquet.py

```diff
@@ -864,6 +864,42 @@ def open_file(path, meta=None):
                                    common_metadata=self.common_metadata)
         return open_file
 
+    def _split_row_groups(self):
+        if not self.metadata or self.metadata.num_row_groups == 0:
+            raise NotImplementedError("split_row_groups is only implemented "
```

Comment:
In future, we should also be able to support this without the central `_metadata` file by loading all the footers of the files in the dataset.

Reply:
Agreed, this was just a first iteration implementing it with minimal effort.